app/src/main/java/org/p2p/wallet/striga/offramp/interactor/StrigaOffRampInteractor.kt

```diff
@@ -0,0 +1,118 @@
+package org.p2p.wallet.striga.offramp.interactor
+
+import timber.log.Timber
+import java.math.BigDecimal
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.minutes
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.launch
+import org.p2p.core.dispatchers.CoroutineDispatchers
+import org.p2p.core.utils.Constants
+import org.p2p.core.utils.isZero
+import org.p2p.core.utils.orZero
+import org.p2p.wallet.infrastructure.coroutines.hasTestScheduler
+import org.p2p.wallet.striga.exchange.models.StrigaExchangeRate
+import org.p2p.wallet.striga.exchange.repository.StrigaExchangeRepository
+import org.p2p.wallet.striga.offramp.models.StrigaOffRampButtonState
+import org.p2p.wallet.striga.offramp.models.StrigaOffRampRateState
+import org.p2p.wallet.striga.offramp.models.StrigaOffRampTokenType
+import org.p2p.wallet.utils.divideSafe
+import org.p2p.wallet.utils.retryOnException
+
+class StrigaOffRampInteractor(
+    private val dispatchers: CoroutineDispatchers,
+    private val strigaExchangeRepository: StrigaExchangeRepository,
+) {
+
+    companion object {
+        private val POLLING_INTERVAL: Duration = 1.minutes
+
+        private val MIN_EUR_AMOUNT = BigDecimal("10")
+        private val MAX_EUR_AMOUNT = BigDecimal("15000")
+    }
+
+    private val exchangeRateState = MutableStateFlow<StrigaOffRampRateState>(StrigaOffRampRateState.Loading)
+
+    private var pollingJob: Job? = null
+
+    fun observeExchangeRateState(): Flow<StrigaOffRampRateState> = exchangeRateState.asStateFlow()
+
+    fun isPollingRunning(): Boolean = pollingJob?.isActive == true
+
+    fun startPolling(scope: CoroutineScope) {
+        if (isPollingRunning()) {
+            Timber.d("Start polling rates: skipped, already running")
+            return
+        }
+        Timber.d("Start polling rates: in progress...")
+        pollingJob = scope.launch(dispatchers.io) {
+            while (isPollingRunning()) {
+                exchangeRateState.emit(StrigaOffRampRateState.Loading)
+
+                try {
+                    retryOnException(exceptionTypes = setOf(Throwable::class)) {
```

Comment:
we removed how many times we can retry?

Reply:
nope, it's already has default arguments:
3 retries; 1000ms interval